core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -0,0 +1,537 @@
+        stopping_round : int, default 10
+            If the validation scores of models fit on pruned data do not improve for stopping_round feature pruning rounds, end the pruning procedure.
+        min_improvement: int, default = 1e-6
+            The newly fitted model's validation score must be >= (1 + min_improvement) * best validation score seen so far for its input
+            feature subset to be considered to be superior to the previous feature subset.
+        max_fits: int, default None
+            If this many models have been fitted during feature pruning, exit the feature pruning loop. Can potentially prevent overfitting.
+            We refit the model using the remaining features after each round of feature pruning.
+
+        Returns
+        -------
+        candidate_features : List[str]
+            Optimal feature subset selected by this method. Set to original features if no features are below pruning threshold or we run
+            out of time before finding a better feature subset.
+        """
+        logger.log(30, f"Performing feature pruning with model: {self.base_model.name}, total time limit: {round(self.time_limit, 2)}s, " +
+                       f"stop threshold: {stopping_round}, prune ratio: {prune_ratio}, prune threshold: {'auto' if not prune_threshold else prune_threshold}.")
+        original_features = X.columns.tolist()
+        if len(original_features) == 1:
+            logger.log(30, f"\tSkipping feature pruning since there is only one feature in the dataset.")
+            return original_features
+        X, y, X_val, y_val, X_fi, y_fi, auto_threshold, subsampled, kwargs = self.setup(X, y, X_val, y_val, n_train_subsample, prune_threshold, kwargs)
+        try:
+            index = 1
+            candidate_features = X.columns.tolist()
+            best_info = {'features': candidate_features, 'index': 1, 'model': self.original_model, 'score': round(self.original_val_score, 4)}
+            if self.model_fit_time is not None and self.time_limit < self.model_fit_time:
+                logger.log(30, f"\tNo time to perform even a single pruning round (remaining: {self.time_limit}, needed: {self.model_fit_time}).")
+                raise TimeLimitExceeded
+
+            # fit self.base_model model once on the subsampled dataset to serve as scoring reference if using subsamples or added a noise column
+            if subsampled or auto_threshold or (self.is_bagged and self.original_model._child_oof):
+                best_info['model'], best_info['score'], _ = self.fit_score_model(X, y, X_val, y_val, candidate_features, f"{self.base_model.name}_1", **kwargs)
+            self._debug_info['index_trajectory'].append(True)
+
+            time_budget_fi = self.compute_time_budget_fi(X_fi=X_fi, n_subsample=n_fi_subsample, **kwargs)
+            logger.log(30, f"\tExpected model fit time: {round(self.model_fit_time, 4)}s, and expected candidate generation time: {round(time_budget_fi, 2)}s.")
```

Comment:
Why 30? That is warning log level. At most this should be 20. Same for several other logs.